crates/biome_js_analyze/src/analyzers/nursery/no_empty_block_statements.rs

```diff
@@ -0,0 +1,176 @@
+    ///
+    /// This rule disallows empty block statements and static blocks. This rule ignores block statements which contain a comment (for example, in an empty catch or finally block of a try statement to indicate that execution should continue regardless of errors).
+    /// This rule also ignores static blocks which contain a comment.
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-empty-static-block/
+    /// Source: https://eslint.org/docs/latest/rules/no-empty/
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// function foo () {}
+    ///
+    /// const foo = () => {}
+    ///
+    /// function fooWithNestedEmptyBlock() {
+    ///     let a = 1;
+    ///     function shouldFail(){}
+    ///     return a
+    ///  }
+    ///
+    /// const fooWithNestedEmptyBlock = () => {
+    ///     let a = 1;
+    ///     const shouldFail = () => {}
+    ///     return a
+    ///  }
+    /// let someVar;
+    /// if (someVar) {
+    /// }
+    ///
+    /// while (someVar) {
+    /// }
+    ///
+    /// switch(someVar) {
+    /// }
+    /// try {
+    ///     doSomething();
+    /// } catch(ex) {
+    ///
+    /// } finally {
+    ///
+    /// }
+    ///
+    // class Foo {
+    //   static {}
+    // }
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// function foo () {let a;}
+    ///
+    /// const foo = () => {let a;}
+    ///
+    /// function fooWithComment() {
+    ///   // should work
+    /// }
+    ///
+    /// const barWithComment = () => {
+    ///   // should work
+    /// }
+    ///
+    /// function fooWithMultilineComment() {
+    ///   /**
+    ///    * this should also work
+    ///    */
+    /// }
+    ///
+    /// const barWithMultilineComment = () => {
+    ///   /**
+    ///    * this should also work
+    ///    */
+    /// }
+    ///
+    ///
+    /// if (foo) {
+    ///   // empty
+    /// }
+    ///
+    /// while (foo) {
+    ///   /* empty */
+    /// }
+    ///
+    /// try {
+    ///   doSomething();
+    /// } catch (ex) {
+    ///   // continue regardless of error
+    /// }
+    ///
+    /// try {
+    ///   doSomething();
+    /// } finally {
+    ///   /* continue regardless of error */
+    /// }
+    ///
+    /// class Foo {
+    ///   static {
+    ///       bar();
+    ///   }
+    /// }
+    ///
+    /// class Foo {
+    ///   static {
+    ///       // comment
+    ///   }
+    /// }
+    /// ```
+    ///
+    pub(crate) NoEmptyBlockStatements {
+        version: "next",
+        name: "noEmptyBlockStatements",
+        recommended: false,
+    }
+}
+
+declare_node_union! {
+    pub(crate) Query = JsBlockStatement | JsFunctionBody | JsStaticInitializationBlockClassMember | JsSwitchStatement
+}
+
+impl Rule for NoEmptyBlockStatements {
+    type Query = Ast<Query>;
+    type State = TextRange;
```

Comment:
Because we use the range of the queried node, I think we can use `()` as state and directly call `query.range()` in `diagnostics` (see proposed changes).

```suggestion
    type State = ();
```